src/fifo/FifoBuffer.cpp

```diff
@@ -31,17 +32,19 @@ FifoBuffer::FifoBuffer(uint32_t bytesPerFrame, uint32_t capacityInFrames)
         : mFrameCapacity(capacityInFrames)
         , mBytesPerFrame(bytesPerFrame)
         , mStorage(NULL)
-        , mReadAtNanoseconds(0)
         , mFramesReadCount(0)
         , mFramesUnderrunCount(0)
         , mUnderrunCount(0)
 {
+    assert(bytesPerFrame > 0);
```

Comment:
Is there a more elegant way we could handle these two assertions?

Reply:
Maybe. I added them because this is a constructor and the FIFO cannot work if the values are incorrect. If they are zero then the FIFO seems to work but doesn't. This took a while to debug. I generally don't like asserts but they seem justified here.

Reply:
Yes agreed, just checking they weren't left in for debugging purposes.